themes/classic/templates/catalog/_partials/product-cover-thumbnails.tpl

```diff
@@ -26,7 +26,7 @@
   {block name='product_cover'}
     <div class="product-cover">
       {if $product.cover}
-        <img class="js-qv-product-cover" src="{$product.cover.bySize.large_default.url}" alt="{$product.cover.legend}" title="{$product.cover.legend}" style="width:100%;" itemprop="image">
+        <img class="js-qv-product-cover" src="{$product.cover.bySize.large_default.url}" alt="{if !empty($product.cover.legend)}{$product.cover.legend}{else}{$product.name|truncate:30:'...'}{/if}" title="{if !empty($product.cover.legend)}{$product.cover.legend}{else}{$product.name|truncate:30:'...'}{/if}" style="width:100%;" itemprop="image">
```

Comment:
Maybe you could add a getter in `ProductLazyArray` to return this text, something like:
```php
public class ProductLazyArray  extends AbstractLazyArray {
    ...
    public function getAlternativeText() {
        if ($this->product->cover) {
            return $this->product->cover;
        }
        
        return Tools::truncate($this->name, 30);
    }
    ...
}
```

Not sure about the method name.

Then you could use something like `alt={$product.alternativeText}` in smarty

Reply:
This would make it easier to use, and maybe there are other images that could need this improve `alt`

Reply:
Look neater thats for sure, also re-use else where would be definitely beneficial. Unfortunately its beyond the scope of my knowledge to implement something like this. PHP isn't my forte neither is the smarty template engine.

Reply:
alt text must be <=125chars.
truncate it to 30 chars  is not good.

The HTML specification does not define a maximum length for "alt" attributes. Current versions of the leading screen reader programs have no limits on the amount of alternate text they will read. However, Freedom Scientific JAWS [1] divides the "alt" attribute into distinct chunks of 125 characters each (excluding spaces), and reads them each separately as if they were separate graphics. This behavior, though it ultimately provides access to the entire "alt" attribute, can be confusing for the user. Also, keep in mind that "alt" attributes additionally benefit sighted users who have images disabled in their browser settings, perhaps in order to minimize download time. For these users, if the height and width of an image is specified, both Internet Explorer and Mozilla truncate the alternate text to fit within the size of the image placeholder. If no image width is specified, Internet Explorer displays the entire "alt" attribute on one line, which may result in a web page that is extraordinarily wide and disorienting.

All of this provides technical justification for keeping the length of "alt" attributes to a minimum, at least shorter than 125 characters in order to avoid JAWS subdividing it. This simply reinforces a common request from screen reader users: Keep alternate text short and sweet. Alternate text should provide equivalent access to the content of an image, but must do so efficiently in order to avoid burdening screen reader users with extraneous information. 

info here: t.ly/Szoi